tests/Backend/ExternalBuckets/BigqueryRegisterBucketTest.php

```diff
@@ -0,0 +1,525 @@
+            $this->assertSame('storage.dbObjectNotFound', $e->getStringCode());
+            $this->assertStringContainsString(
+                'Could not map bindings for google.cloud.bigquery.analyticshub.v1.AnalyticsHubService/SubscribeListing to any Uri template.',
+                $e->getMessage()
+            );
+        }
+    }
+
+    public function testRegisterWSAsExternalBucket(): void
+    {
+        $description = $this->generateDescriptionForTestObject();
+        $testBucketName = $this->getTestBucketName($description);
+        $bucketId = self::STAGE_IN . '.' . $testBucketName;
+
+        $this->dropBucketIfExists($this->_client, $bucketId, true);
+        $this->initEvents($this->_client);
+
+        $externalBucketBackend = 'bigquery';
+        $guide = $this->_client->registerBucketGuide(['external_bucket'], $externalBucketBackend);
+        $this->assertArrayHasKey('markdown', $guide);
+
+        // prepare external bucket
+        $createdListing = $this->prepareExternalBucketForRegistration($description);
+
+        // register external bucket
+        $runId = $this->setRunId();
+        $idOfBucket = $this->_client->registerBucket(
+            $testBucketName,
+            [$createdListing->getName()],
+            'in',
+            'Iam in external bucket',
+            $externalBucketBackend,
+            'Iam-your-external-bucket'
+        );
+
+        $assertCallback = function ($events) {
+            $this->assertCount(1, $events);
+        };
+        $query = new EventsQueryBuilder();
+        $query->setEvent('storage.bucketCreated')
+            ->setTokenId($this->tokenId)
+            ->setRunId($runId);
+        $this->assertEventWithRetries($this->_client, $assertCallback, $query);
+
+        // check external bucket
+        $bucket = $this->_client->getBucket($idOfBucket);
+        $this->assertTrue($bucket['hasExternalSchema']);
+
+        $tables = $this->_client->listTables($idOfBucket);
+        $this->assertCount(0, $tables);
+        // check external bucket
+
+        $externalCredentials['connection']['backend'] = 'bigquery';
+        $externalCredentials['connection']['credentials'] = $this->getCredentialsArray();
+        $externalCredentials['connection']['schema'] = sha1($description). '_external_bucket';
+
+        // add first table to external bucket
+        // I created a user for the external bucket the same way as for WS.
+        // Workspace can't just be used because the user doesn't have the right to create the exchanger and the listing
+        $db = WorkspaceBackendFactory::createWorkspaceBackend($externalCredentials);
+        $db->createTable('TEST', [
+            'AMOUNT' => 'INT',
+            'DESCRIPTION' => 'STRING',
+        ]);
+        $db->executeQuery(sprintf(
+        /** @lang BigQuery */
+            'INSERT INTO `%s`.`TEST` (`AMOUNT`, `DESCRIPTION`) VALUES (1, \'test\');',
+            $externalCredentials['connection']['schema']
```

Comment:
Tady by měl být quote, ale možná to není potřeba přímo tady

Reply:
jo ono tam ten quote je priamo v tom stringu ale upravil som to

Reply:
Povedzde, Kefalín, čo vy si představujete pod tým pojmom "quoting"? :D Co když bude schema name 
```
my` VALUES (1, 'test'); DROP TABLE TEST; #
```
To by ti ty quotes ve stringu moc nepomohly :) 

Ale rozumím, že tady není to schéma z uživatelského vstupu...